src/server/scripts/Spells/spell_generic.cpp

```diff
@@ -4460,6 +4460,33 @@ class spell_gen_impatient_mind : public SpellScriptLoader
         }
 };
 
+// 180457 - Unholy Resurrection.
+class spell_gen_rasthe_unholy_resurrection : public SpellScriptLoader
+{
+public:
+    spell_gen_rasthe_unholy_resurrection() : SpellScriptLoader("spell_gen_rasthe_unholy_resurrection") { }
+
+    class spell_gen_rasthe_unholy_resurrection_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_gen_rasthe_unholy_resurrection_SpellScript);
+
+        void HandleScript(SpellEffIndex /*effIndex*/)
+        {
+             if (Unit* target = GetHitUnit())
```

Comment:
@Ulduar right, spell testing in game and working.
@Keader spell has target 1 (UNIT_CASTER). In this case better add not null for hit object.